src/DynamicRupture/FrictionLaws/ImposedSlipRates.h

```diff
@@ -41,15 +39,19 @@ class ImposedSlipRates : public BaseFrictionLaw<ImposedSlipRates<STF>> {
     for (unsigned pointIndex = 0; pointIndex < misc::numPaddedPoints; pointIndex++) {
       real stfEvaluated = stf.evaluate(currentTime, timeIncrement, ltsFace, pointIndex);
 
-      tractionResults.traction1[timeIndex][pointIndex] =
-          faultStresses.traction1[timeIndex][pointIndex] -
-          this->impAndEta[ltsFace].etaS * slip1[ltsFace][pointIndex] * stfEvaluated;
-      tractionResults.traction2[timeIndex][pointIndex] =
-          faultStresses.traction2[timeIndex][pointIndex] -
-          this->impAndEta[ltsFace].etaS * slip2[ltsFace][pointIndex] * stfEvaluated;
+      this->traction1[ltsFace][pointIndex] = faultStresses.traction1[timeIndex][pointIndex] -
+                                             this->impAndEta[ltsFace].etaS *
+                                                 imposedSlipDirections[ltsFace][pointIndex][0] *
+                                                 stfEvaluated;
+      this->traction2[ltsFace][pointIndex] = faultStresses.traction2[timeIndex][pointIndex] -
+                                             this->impAndEta[ltsFace].etaS *
+                                                 imposedSlipDirections[ltsFace][pointIndex][1] *
```

Comment:
I guess if we re-write it as follows  `imposedSlipDirections[ltsFace][0][pointIndex]` and `imposedSlipDirections[ltsFace][1][pointIndex]` it will result in a better vectorization. Moreover, it will become GPU-friendly i.e., we would share more data structures